src/Illuminate/Foundation/EnvironmentDetector.php

```diff
@@ -48,7 +48,10 @@ protected function detectConsoleEnvironment(Closure $callback, array $args)
         // and if it was that automatically overrides as the environment. Otherwise, we
         // will check the environment as a "web" request like a typical HTTP request.
         if (! is_null($value = $this->getEnvironmentArgument($args))) {
-            return head(array_slice(explode('=', $value), 1));
+            $env = head(array_slice(explode('=', $value), 1));
+            putenv("APP_ENV=$env");
```

Comment:
Feels wrong to call that function out of the blue. We never call that function in laravel at all. I just did a search. The environment detector is meant to detect the environment, not modify it.